romanesco/__init__.py

```diff
@@ -132,8 +131,7 @@ def convert(type, input, output, **kwargs):
         returns the output binding unchanged.
     """
 
-    if "data" not in input:
-        input["data"] = romanesco.io.fetch(input, **kwargs)
+    input["data"] = romanesco.io.fetch(input, **kwargs)
```

Comment:
I have a feeling that may be the case. If so, we should add another key into the input dict like `fetched` or something to indicate that it has already been fetched.